### PR TITLE
Close Can Streams whenever one is opened

### PR DESCRIPTION
**Pull request by SenseiGia**
Fix  https://github.com/CrossTheRoadElectronics/Phoenix-IssueTracker/issues/408

+ Add CloseStream() to ConfigSet and ConfigGet
+ Add retry logic for CAN.Send()
+ Only Open Stream when required
+ Minor white space fix

### CODE REVIEW

**Comment by ozrien** [comment]
@SenseiGia  I'm having trouble following the diffs.  Also it's conflicted anyway.  Gia can you resubmit with a cleaner version.  Just change the lines of code that matter.  There's a lot a whitespace noise and a comment that references another feature.

File: CTRE/LowLevel/Device_LowLevel.cs [hunk 1]
```diff
@@ -348,9 +345,22 @@ private ErrorCode ConfigSetParameterRaw(ParamEnum paramEnum, int value, byte sub
             /* send it */
             uint arbId = PARAM_SET | GetDeviceNumber();
             int status = CTRE.Native.CAN.Send(arbId, frame, 8, 0);
+            /* Send Error Exists when Timeout is 0ms and doing continuous configs, or high CAN Utiliztion */
+            if (status != 0)
+            {
+                /* Retry until success or we have exceeded 1000 attempts. */
+                for (int i = 0; i < 1000; i++)
```
Comment (ozrien) [inline CTRE/LowLevel/Device_LowLevel.cs]: 1000 is crazy, how about a max of 5?

Comment (SenseiGia) [reply CTRE/LowLevel/Device_LowLevel.cs]: At 5 attempts, we fail after about every 300 Configs, returning a -1 ErrorCode. I'm not sure if we want to recreate a more realistic scenario and be fine with that or test worse case and have some massive number (my 1000) to always succeed at the CAN.Send

Comment (ozrien) [reply CTRE/LowLevel/Device_LowLevel.cs]: I added a sleep(1) in my version - that should be enough time (5-ms) for a frame slot to become free.  I'm not sure how much time 1000 fast calls are.

**Comment by ozrien** [comment]
Fresh PR is [here](https://github.com/CrossTheRoadElec/Phoenix-netmf-lib/pull/22)

**Comment by JCaporuscio** [comment]
Closing PR since changes have been merged as part of https://github.com/CrossTheRoadElec/Phoenix-netmf-lib/pull/22

**Pull request closed without merging**